src/discordcr/rest.cr

```diff
@@ -335,20 +336,30 @@ module Discord
     # [API docs for this method](https://discordapp.com/developers/docs/resources/channel#create-message)
     # (same as `#create_message` -- this method implements form data bodies
     # while `#create_message` implements JSON bodies)
-    def upload_file(channel_id : UInt64, content : String?, file : File)
+    def upload_file(channel_id : UInt64, content : String?, file : IO, filename : String? = nil)
       io = IO::Memory.new
-      boundary = "--------------------------#{SecureRandom.urlsafe_base64(18)}"
-      HTTP::FormData.generate(io, boundary) do |g|
-        g.field("content", content) if content
-        g.file("file", file, HTTP::FormData::FileMetadata.new(filename: File.basename(file.path)))
+
+      if filename.nil?
+        if file.is_a? File
+          filename = File.basename(file.path)
+        else
+          filename = ""
```

Comment:
If it is `nil` the sending will not work at all as Discord expects the property to be sent; if it's an empty string a placeholder filename will be generated.